### PR TITLE
Fixes potential NPE when custom converters receive empty parameters

### LINKED ISSUE #40287: Using a custom ParamConverter fails when not sending that parameter with the request
### Describe the bug

When using a custom ParamConverter, the call fails when we do not sending  any value for that parameter in the request.

### Expected behavior

The request would be executed

### Actual behavior

A 404 error happens due to a NPE.

### How to Reproduce?

Take this example code:

```
@GET
    @Path("/greet")
    public Response greet(@QueryParam("number") @Parameter(allowEmptyValue = true) Optional<Integer> numberOpt) {
        if (numberOpt.isPresent()) {
            Integer number= numberOpt.get();
            if (number == null) {
                return Response.ok("Hello! You provided an empty number.").build();
            } else {
                return Response.ok("Hello, " + number+ "!").build();
            }
        } else {
            return Response.ok("Hello, world! No number was provided.").build();
        }
    }
```
    
This is the converter:

```
public class OptionalIntegerParamConverter implements ParamConverter<Optional<Integer>> {

	@Override
    public Optional<Integer> fromString(String value) {
		if (value == null) {
			return null;
		}
		
        if (value.trim().isEmpty()) {
        	return Optional.empty();
        }
        
        try {
        	int parsedInt = Integer.parseInt(value);
        	return Optional.of(parsedInt);
        } catch (NumberFormatException e) {
            throw new IllegalArgumentException("Invalid integer value");
        }
    }
    @Override
    public String toString(Optional<Integer> value) {
    	if (!value.isPresent()) {
    		return null;
    	}
    	
    	Integer intValue = value.get();
        if (intValue == null) {
            return null;
        } else {
            return intValue.toString();
        }
    }

}

}
```
And this is its provider:
```
@Provider
public class OptionalIntegerParamConverterProvider implements ParamConverterProvider {
    @SuppressWarnings("unchecked")
	@Override
    public <T> ParamConverter<T> getConverter(Class<T> rawType, Type genericType, Annotation[] annotations) {
    	if (rawType.equals(Optional.class)) {
            if (genericType instanceof ParameterizedType) {
                ParameterizedType parameterizedType = (ParameterizedType) genericType;
                Type[] typeArguments = parameterizedType.getActualTypeArguments();
                if (typeArguments.length == 1 && typeArguments[0].equals(Integer.class)) {
                    return (ParamConverter<T>) new OptionalIntegerParamConverter();
                }
            }
        }
        
        return null; // return null if no converter is needed for the type
    }
}
```

If that is setup, then trying to call the service without providing any value to the number parameter will fail within `RuntimeResolvedConverter` 

```
public Object convert(Object parameter) {
        if (runtimeConverter != null)
            return runtimeConverter.fromString(parameter.toString());
        return quarkusConverter.convert(parameter);
    }
```
it fails because parameter is null.

If I provide no conversion, then it fails only when checking the `Send empty value`, since "" cannot be converted to an Optional<Integer>

Thus, a simple change in `RuntimeResolvedConverter` would fix this:
```
public Object convert(Object parameter) {
        if (runtimeConverter != null)
            return runtimeConverter.fromString(parameter != null ? parameter.toString() : null);
        return quarkusConverter.convert(parameter);
    }
```

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by gastaldi**
- Fixes #40287

### CODE REVIEW

File: integration-tests/pom.xml [hunk 1]
```diff
@@ -332,6 +332,7 @@
                 <module>reactive-messaging-rabbitmq-dyn</module>
                 <module>reactive-messaging-hibernate-reactive</module>
                 <module>reactive-messaging-hibernate-orm</module>
+                <module>rest</module>
```
Comment (geoand) [inline integration-tests/pom.xml:335]: I don't think should add a new module for this

Comment (gastaldi) [reply integration-tests/pom.xml:335]: Can you point me to where should I put it then?

Comment (geoand) [reply integration-tests/pom.xml:335]: Unless I am missing something, it can be a QuarkusUnitTest in the RESTEasy Reactive server's deployment module

**Comment by michelcouto** [comment]
> /cc @michelcouto

Thanks!

**Review by geoand — APPROVED** [review]

**Pull request merged**